### PR TITLE
186719010 fix stylings

### PR DESCRIPTION
**Pull request by eireland**
Fixes column, device, and arrow styling to match spec

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -42,6 +46,10 @@ export const App = () => {
     setModel({columns: [{devices: [createDefaultDevice()]}]});
   }, [setModel]);
 
+  useEffect(()=>{
+    setSelectedDeviceId(lastDeviceId);
```
Comment (dougmartin) [inline src/components/App.tsx:50]: Does this cause the last device to be always selected when any change is made to the model?

File: src/components/model/arrow.tsx [hunk 2]
```diff
@@ -127,29 +127,31 @@ export const Arrow = ({source, target, model, selectedDeviceId}: IProps) => {
...
   const targetRect = getRect(targetDiv);
+  const sourceMidPointY = (sourceRect.bottom - sourceRect.top) / 2;
+
 
-  let svgTop = Math.min(sourceRect.midY, targetRect.midY);
+  let svgTop = Math.min(sourceMidPointY, targetRect.midY);
   let svgBottom = Math.max(sourceRect.midY, targetRect.midY);
   let svgHeight = svgBottom - svgTop;
-  const horizontalArrow = svgHeight === 0;
+  const horizontalArrow = svgHeight <= 164;
```
Comment (dougmartin) [inline src/components/model/arrow.tsx:136]: Why 164? Maybe this should be a constant?

File: src/components/model/arrow.tsx [hunk 3]
```diff
@@ -127,29 +127,31 @@ export const Arrow = ({source, target, model, selectedDeviceId}: IProps) => {
...
   if (svgHeight < minSvgHeight) {
-    const halfHeightDiff = (minSvgHeight - svgHeight) / 2;
+    const halfHeightDiff = (minSvgHeight + svgHeight) / 2;
     svgHeight = minSvgHeight;
     svgTop -= halfHeightDiff;
     svgBottom += halfHeightDiff;
   }
 
-  const svgLeft = sourceRect.right;
+  const svgLeft = -40;
```
Comment (dougmartin) [inline src/components/model/arrow.tsx:145]: Why -40?  Maybe this should be a constant?

File: src/components/model/arrow.tsx [hunk 4]
```diff
@@ -127,29 +127,31 @@ export const Arrow = ({source, target, model, selectedDeviceId}: IProps) => {
...
-
   const start: IPoint = {x: 0, y: sourceRect.midY - svgTop};
   const end: IPoint = {x: svgWidth, y: targetRect.midY - svgTop};
 
-  const svgStyle: React.CSSProperties = {top: svgTop, left: svgLeft, width: svgWidth, height: svgHeight};
+  const arrowMidPoint = (end.y - start.y) / 2;
+  const labelTop = horizontalArrow ? 0 : arrowMidPoint < 0 ? arrowMidPoint - kMaxLabelHeight/2 : -arrowMidPoint - kMaxLabelHeight;
+  const labelLeft = (svgWidth / 2) - (labelDivWidth / 2);
+
+  const arrowContainerStyle: React.CSSProperties = {top: 0, left: svgLeft, width: svgWidth, height: svgHeight + 10};
```
Comment (dougmartin) [inline src/components/model/arrow.tsx:154]: Why +10?

File: src/components/model/column.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import React, {useEffect} from "react";
```
Comment (dougmartin) [inline src/components/model/column.tsx:1]: We should clean these unused imports up.

File: src/components/model/device.tsx [hunk 6]
```diff
@@ -25,7 +25,8 @@ interface IProps {
 }
 
 export const Device = (props: IProps) => {
-  const {model, device, selectedDeviceId, setSelectedDeviceId, addDevice, mergeDevices, deleteDevice, handleNameChange, handleInputChange} = props;
+  const {model, device, selectedDeviceId, setSelectedDeviceId, addDevice, mergeDevices, deleteDevice,
+    handleNameChange, handleInputChange} = props;
```
Comment (dougmartin) [inline src/components/model/device.tsx:29]: Need to remove `handleNameChange` to remove the linting error.

**Review by dougmartin — APPROVED** [review]
Looks good :+1: 

Just some suggestions to clean up the lint errors and add some constants.

File: src/components/model/arrow.tsx [hunk 7]
```diff
@@ -127,29 +127,31 @@ export const Arrow = ({source, target, model, selectedDeviceId}: IProps) => {
 
   const sourceRect = getRect(sourceDiv);
   const targetRect = getRect(targetDiv);
+  const sourceMidPointY = (sourceRect.bottom - sourceRect.top) / 2;
```
[the remark this answers is not in the document]
Comment (eireland) [reply src/components/model/arrow.tsx:130]: `midY` takes into account the rect y position. This calc does not.

File: src/components/App.tsx (hunk 1 above)
Comment (eireland) [reply src/components/App.tsx:50]: I have not noticed in testing.

File: src/components/model/arrow.tsx (hunk 2 above)
Comment (eireland) [reply src/components/model/arrow.tsx:136]: Changed logic to test if source and target are at the same midY

**Pull request merged**